File: sample/src/main/java/me/drakeet/multitype/sample/payload/TestPayloadActivity.java
/*
 * Copyright 2016 drakeet. https://github.com/drakeet
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package me.drakeet.multitype.sample.payload;

import android.os.Bundle;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.RecyclerView;
import android.widget.Toast;
import me.drakeet.multitype.Items;
import me.drakeet.multitype.MultiTypeAdapter;
import me.drakeet.multitype.sample.MenuBaseActivity;
import me.drakeet.multitype.sample.R;

import static android.support.v7.widget.DividerItemDecoration.VERTICAL;

public class TestPayloadActivity extends MenuBaseActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_list);
    RecyclerView recyclerView = findViewById(R.id.list);
    recyclerView.addItemDecoration(new DividerItemDecoration(this, VERTICAL));
    MultiTypeAdapter adapter = new MultiTypeAdapter();
    recyclerView.setAdapter(adapter);

    adapter.register(HeavyItem.class, new HeavyItemViewBinder());

    Items items = new Items();
    for (int i = 0; i < 30; i++) {
      items.add(new HeavyItem("1000" + i));
    }
    adapter.setItems(items);
    adapter.notifyDataSetChanged();

    Toast.makeText(this, "Try to click or long click items", Toast.LENGTH_SHORT).show();
  }
}

File: sample/src/main/kotlin/com/drakeet/multitype/sample/payload/TestPayloadActivity.kt
/*
 * Copyright (c) 2016-present. Drakeet Xu
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.drakeet.multitype.sample.payload

import android.os.Bundle
import android.widget.Toast
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.DividerItemDecoration.VERTICAL
import androidx.recyclerview.widget.RecyclerView
import com.drakeet.multitype.MultiTypeAdapter
import com.drakeet.multitype.sample.MenuBaseActivity
import com.drakeet.multitype.sample.R
import java.util.*

class TestPayloadActivity : MenuBaseActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_list)
    val recyclerView = findViewById<RecyclerView>(R.id.list)
    recyclerView.addItemDecoration(DividerItemDecoration(this, VERTICAL))
    val adapter = MultiTypeAdapter()
    recyclerView.adapter = adapter

    adapter.register(HeavyItemViewBinder())

    val items = ArrayList<Any>()
    for (i in 0..29) {
      items.add(HeavyItem("1000$i"))
    }
    adapter.items = items
    adapter.notifyDataSetChanged()

    Toast.makeText(this, "Try to click or long click items", Toast.LENGTH_SHORT).show()
  }
}
